src/Api/Test/ClientTest.php

```diff
@@ -0,0 +1,572 @@
+            'json' => [
+                'description' => 'description',
+                'isActive' => true,
+                'maxParallelJobs' => 1,
+                'name' => 'client'.$timestamp,
+                'owner' => 1,
+                'quota' => -1,
+                'url' => 'root@172.17.0.1'
+            ]
+        ]);
+        $iri = $this->findIriBy(Client::class, ['name' => 'client'.$timestamp]);
+        
+        $httpClient->request('PUT', $iri, [
+            'json' => [
+                'description' => 'description updated',
+                'isActive' => true,
+                'maxParallelJobs' => -1,
+                'name' => 'client'.$timestamp,
+                'owner' => 1,
+                'quota' => -1,
+                'url' => 'root@172.17.0.1'
+            ]
+        ]);
+        $this->assertResponseStatusCodeSame(422);
+        $this->assertResponseHeaderSame('content-type', 'application/ld+json; charset=utf-8');
+        $this->assertJsonContains([
+            '@context' => '/api/contexts/ConstraintViolationList',
+            '@type' => 'ConstraintViolationList',
+            'hydra:title' => 'An error occurred',
+            'hydra:description' => 'maxParallelJobs: Max parallel jobs value must be a positive integer',
+        ]);
+    }
+    
+    public function testUpdateClientInvalidName(): void
+    {
+        $httpClient = $this->authenticate();
+        $timestamp = $this->getTimestamp();
+        $httpClient->request('POST', '/api/clients', [
+            'json' => [
+                'isActive' => true,
+                'maxParallelJobs' => 1,
+                'name' => 'clientToRepeat',
+                'owner' => 1,
+                'quota' => -1
+            ]
+        ]);
+        $httpClient->request('POST', '/api/clients', [
+            'json' => [
+                'description' => 'description',
+                'isActive' => true,
+                'maxParallelJobs' => 1,
+                'name' => 'clientToUpdate',
+                'owner' => 1,
+                'quota' => -1,
+                'url' => 'root@172.17.0.1'
+            ]
+        ]);
+        $iri = $this->findIriBy(Client::class, ['name' => 'clientToUpdate']);
+        
+        $httpClient->request('PUT', $iri, [
+            'json' => [
+                'isActive' => true,
+                'maxParallelJobs' => 1,
+                'name' => 'clientToRepeat',
+                'owner' => 1,
+                'quota' => -1
+            ]
+        ]);
+        $this->assertResponseStatusCodeSame(400);
+        $this->assertJsonContains([
+            '@context' => '/api/contexts/Error',
+            '@type' => 'hydra:Error',
+            'hydra:title' => 'An error occurred',
+        ]);
+    }
+    
+    public function testUpdateClientNotFound(): void
+    {
+        $httpClient = $this->authenticate();
+        $timestamp = $this->getTimestamp();
+        $httpClient->request('PUT', '/api/clients/0', [
```

Comment:
Hi number instead ?
We are using the UNEXISTING_ID many times so maybe we should take it to a constant.